### PR TITLE
Revert "entropy: cc310: Use cc310 with MPU"

### PR DESCRIPTION
**Pull request by thst-nordic**
- fix desktop dfu test failure
This reverts commit 44a52c1e10463eb6069ff7077e7005e96e89bd68.

### CODE REVIEW

File: drivers/entropy/Kconfig [hunk 1]
```diff
@@ -11,6 +11,7 @@ config ENTROPY_CC310
 	depends on ENTROPY_GENERATOR
 	depends on !BT_LL_SW_LEGACY
 	depends on !BT_LLL_VENDOR_NORDIC
+	depends on !MPU_STACK_GUARD
```
Comment (tejlmand) [inline drivers/entropy/Kconfig]: This was a temporary fix at the time.
The correct fix is this: https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/1356/files

Comment (tejlmand) [reply drivers/entropy/Kconfig]: apparently this file lacks the disabling of cc310:
https://github.com/NordicPlayground/fw-nrfconnect-nrf/blob/f3859825e2bba3f696bf9fd8e1b225c714851bdf/applications/nrf_desktop/configuration/nrf52840_pca20041/app_ZDebugMCUBoot.conf#L248

Comment (tejlmand) [reply drivers/entropy/Kconfig]: Added backport label to this PR:
https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/1369

**Comment by thst-nordic** [comment]
closing as duplicate #1471

**Pull request closed without merging**